### ISSUE TITLE
Issue with creating variants from different scenarios

### ISSUE THREAD

**Issue by JeanLucEsser** [nystudio107/craft3-imageoptimize#335]
So to continue our discussion from Discord.

I switched to S3 and AWS Sharp Image Transforms.
All images are still pointing to the local transform volume.
If I go to an asset and resave it, variants are generated and everything works fine.
So the logic next step is to make sure to re create all variants.

I tried to save a global as it is supposed to kick off a full resave but nothing happened (except for a quick search index job). Nothing related to ImageOptimize in queue.log either.

I tried to do a imageoptimize/optimize/create --force via the CLI but still nothing. Nothing related to ImageOptimize in console.log either.

So I tried to update asset indexes in the CP and that did start more than a thousand jobs (one per variant). A couple timeout errors later, all jobs finished and all variants are now generated. queue.log is chockfull of all the variants generation, no errors are logged.

But if I hear you correctly, that's not how it is supposed to work. And saving a global still doesn't kick off anything. CLI commands neither.

If I edit a volume and save it though, it does generate a single queue job for that volume.

So to sum it up:
- saving a global does not trigger anything, no errors logged
- CLI command does not create all variants, no errors logged
- updating assets via CP does trigger one job per variant (thousands of jobs)
- editing and saving a volume does the correct thing, one job for the volume

Craft 4.0.3 / Plugin beta 3

**Comment by JeanLucEsser** [comment]
FWIW, this does not seem to be related to Sharp, I tried switching to CraftTransforms and the same applies. Volumes are still in a S3 Bucket though, only the volume dedicated to transforms is local.

**Comment by JeanLucEsser** [comment]
Just to follow up on this.

I don't care if updating asset indexes does trigger one job per variant, unless if you say it shouldn't. As long as we're using async queue, it's a non blocking issue (when it'll work as async queue has its own issues right now ; see https://github.com/ostark/craft-async-queue/issues/55).

The CLI commands not working is troubling but maybe I need to investigate some more, could be on my end. Same for updating via saving a global.

For now I managed to regenerate all variants, and I'll use async queue when the issue gets resolved.

Anyway, tell me how I can help you with this should you need it.

**Comment by khalwat** [comment]
Will be in touch as I investigate it.

**Comment by khalwat** [comment]
Addressed in: https://github.com/nystudio107/craft-imageoptimize/commit/ef806e75486504621176a2183325697efce68845

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-imageoptimize": "dev-develop-v4 as 4.0.0-beta.4”,
```
Then do a `composer clear-cache && composer update`
